docs/docs/adding-react-components.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Adding React Components
+---
+
+This guide covers how to add React components to your Gatsby site.
+
+## React components
+
+React components are prebuilt elements or groups of elements that can be used to prevent repeating yourself in your code.
+
+Components also offer the ability to be customised using "props" (properties). Props can be of any Javascript type, such as boolean, string, object, array or almost anything you can think of.
+
+For example, you could use a component for Buttons on your site. This is because they would likely be used multiple times across pages and would need different labels each time.
+
+## Importing React components
+
+In Gatsby, when using React components, you import and use them just like you would in normal React:
+
+```jsx
+import React from "react"
+import { Link } from "gatsby"
+
+export default () => (
+  <div>
+    <Link to="/contact/">Contact</Link>
+  </div>
+)
+```
+
```

Comment:
The issue mentions [adding in 3rd party libraries](https://github.com/gatsbyjs/gatsby/issues/16288#issue-475710777), perhaps we could also add in an example - after this one about your own component - to importing a 3rd party component library like Bootstrap, React Select, or really anything that could be installed from npm and then used in a Gatsby site.

Reply:
This would be a nice opportunity to explain how to write a component. I have a few links and examples in my Gatsby workshop:
- https://marcysutton.github.io/gatsby-a11y-workshop/slides/29
- https://marcysutton.github.io/gatsby-a11y-workshop/slides/30
- https://marcysutton.github.io/gatsby-a11y-workshop/slides/31

The people who would need the information that's in this doc may not already know how to write a component. Consuming them is an important thing to cover, and this doc does that really well! But filling it in a bit more with React component information would be excellent. It would also provide a nice setup for the Porting from Create React App to Gatsby doc.

Reply:
That's a good idea. Isn't there already documentation for that on the React site, though? It's it worth making a (not completely, but basically) copy and paste version from their docs?

Reply:
That makes sense -- there is a great doc from Reactjs.org that I've added in a suggestion in the top paragraph.